src/Hazelcast.Net.Tests/Clustering/FailoverTests.cs

```diff
@@ -80,74 +76,93 @@ public void TestClusterOptionsRotate()
             {
-                Assert.AreEqual(clusterName1, failover.CurrentClusterOptions.ClusterName);
-                Assert.True(failover.CurrentClusterOptions.Networking.Addresses.Contains(address1));
+                Assert.AreEqual(clusterName1, fo.CurrentClusterOptions.ClusterName);
+                Assert.True(fo.CurrentClusterOptions.Networking.Addresses.Contains(address1));
             };
 
-            void assertForCluster2(Failover failover)
+            void AssertCluster2(Failover fo)
             {
-                Assert.AreEqual(clusterName2, failover.CurrentClusterOptions.ClusterName);
-                Assert.True(failover.CurrentClusterOptions.Networking.Addresses.Contains(address2));
-                Assert.False(failover.CurrentClusterOptions.Networking.Addresses.Contains(address1));
-                Assert.AreEqual(failover.CurrentClusterOptions.Authentication.CredentialsFactory.Service.NewCredentials().Name, username);
+                Assert.AreEqual(clusterName2, fo.CurrentClusterOptions.ClusterName);
+                Assert.True(fo.CurrentClusterOptions.Networking.Addresses.Contains(address2));
+                Assert.False(fo.CurrentClusterOptions.Networking.Addresses.Contains(address1));
+                Assert.AreEqual(fo.CurrentClusterOptions.Authentication.CredentialsFactory.Service.NewCredentials().Name, username);
             };
 
-            //initial one must be cluster 1
-            assertForCluster1(failover);
+            // initial one must be cluster 1
+            AssertCluster1(failover);
             clusterState.ChangeState(ClientState.Disconnected);
 
-            //Loop 1, Try 1
-            failover.RequestClusterChange();
-            assertForCluster2(failover);
-            Assert.AreEqual(1, countOfClusterChangedRaised);
-            Assert.AreEqual(0, failover.CurrentTryCount);
-
-            //Loop 1, Try 2
-            failover.RequestClusterChange();
-            assertForCluster1(failover);
-            Assert.AreEqual(2, countOfClusterChangedRaised);
-            Assert.AreEqual(1, failover.CurrentTryCount);
-
-            //Loop 2, Try 1
-            failover.RequestClusterChange();
-            assertForCluster2(failover);
-            Assert.AreEqual(3, countOfClusterChangedRaised);
-            Assert.AreEqual(1, failover.CurrentTryCount);
-
-            //Loop 2, Try 2
-            failover.RequestClusterChange();
-            assertForCluster1(failover);
-            Assert.AreEqual(4, countOfClusterChangedRaised);
-            Assert.AreEqual(2, failover.CurrentTryCount);
-
-            //Loop 3-> spent all tries
-            Assert.False(failover.RequestClusterChange());
+            var expectedCount = 1;
+
+            // Loop 1, Try 1
+            Assert.That(failover.TryNextCluster());
+            AssertCluster2(failover);
+            Assert.AreEqual(expectedCount, clusterChangedCount);
+            Assert.AreEqual(expectedCount, failover.CurrentTryCount);
```

Comment:
This could fail since `failover.CurrentTryCount` is incremented after a loop is completed, not on each cluster change.

Reply:
Okay, I have just seen the `Failover`. So, this test should work.